onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -2686,6 +2686,149 @@ It's an extension of Gelu. It takes the sum of input A and bias input B as the i
         updateOutputElemType(ctx, 0, ONNX_NAMESPACE::TensorProto::BOOL);
       });
 
+  static const char* OptionalEmpty_ver1_doc = R"DOC(
+      Construct an empty optional, with given element type.
+      )DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(OptionalEmpty)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc(OptionalEmpty_ver1_doc)
+      .Attr("type", "Type of the element in the optional output", AttributeProto::TYPE_PROTO)
+      .Output(
+          0,
+          "output",
+          "An optional that contains no elements.",
+          "O")
+      .TypeConstraint(
+          "O",
+          {"optional(tensor(float))",
+           "optional(seq(tensor(float)))"},
+          "Constrains output type to all optional tensor or optional sequence types.")
```

Comment:
It is not really all optional tensors right? We are restricting it to tensor of type float and optional sequence of tensor of type float... tensor of type double wont work

Reply:
I am planning to add all the types in the standard spec. Kept float only here for simplicity. Will extend it.